internal/app/coreValues/service.go

```diff
@@ -112,13 +122,57 @@ func (cs *service) UpdateCoreValue(ctx context.Context, coreValueID string, reqD
 		return
 	}
 
-	resp, err = cs.coreValuesRepo.UpdateCoreValue(ctx, coreValId, reqData)
+	reqData.Id = coreValId
+
+	dbResp, err := cs.coreValuesRepo.UpdateCoreValue(ctx, reqData)
 	if err != nil {
-		logger.WithField("err", err.Error()).Error("Error while updating core value")
 		err = apperrors.InternalServerError
 
 		return
 	}
 
+	resp = MapCoreValueDbToService(dbResp)
+
+	return
+}
+
+func (cs *service) ValidateParentCoreValue(ctx context.Context, coreValueID int64) (ok bool) {
+	coreValue, err := cs.coreValuesRepo.GetCoreValue(ctx, coreValueID)
+	if err != nil {
+		logger.WithField("err", err.Error()).Error("Parent core value id not present")
+		return
+	}
+
+	if coreValue.ParentCoreValueID.Valid {
+		logger.Error("Invalid parent core value id")
+		return
+	}
+
+	return true
+}
+
+func (cs *service) Validate(ctx context.Context, coreValue dto.CreateCoreValueReq) (err error) {
+
+	if coreValue.Name == "" {
+		err = apperrors.TextFieldBlank
+	}
+	if coreValue.Description == "" {
+		err = apperrors.DescFieldBlank
+	}
+	if coreValue.ParentCoreValueID != nil {
+		svc := NewService(cs.coreValuesRepo)
+		if !svc.ValidateParentCoreValue(ctx, *coreValue.ParentCoreValueID) {
+			err = apperrors.InvalidParentValue
+		}
+	}
+
+	return
+}
+
+func MapCoreValueDbToService(dbStruct repository.CoreValue) (svcStruct dto.CoreValue) {
```

Comment:
can make it unexported, if no other component is directly calling corevalues' DB methods.

Reply:
Done